pitch.md

```diff
@@ -23,38 +51,192 @@ should be more specific than "The user describes properties of unicorns.", e.g.,
 by describing which unicorn properties (horn color, tattoo, etc.) are most 
 important to users and what kinds of values those properties might have._ 
 
+The user will be able to define different rules that specify the coordinate where a 
+line starts, the direction it travels towards, and the distance it travels. Optional 
+things that the user can define is the speed of the line, the weight of the line, the
+color of the line, and whether or not the line will disappear as it travels (otherwise 
+default values will be set). Here is an example of the code I am envisioning:
+
+```
+// Creates two small spirals at the same time
+Start (50, 50) 
+Right: 10
+Down: 10
+Left: 15
+Up: 15
+Right: 20
+Down: 20
+Left: 25
+Up: 25
+
+Start (100, 50)
+Right: 10
+Down: 10
+Left: 15
+Up: 15
+Right: 20
+Down: 20
+Left: 25
+Up: 25
+```
+
+But I also want the user to be able to store and reuse a set of rules, so the code could be rewritten as follows:
+
+```
+// Creates two small spirals
+Rule spiralRule = 
+{
+	Right: 10
+	Down: 10
+	Left: 15
+	Up: 15
+	Right: 20
+	Down: 20
+	Left: 25
+	Up: 25
+}
+
+Start (50, 50)
+Do spiralRule
+	
+Start (100, 50)
+Do spiralRule
+```
+
+Finally, here is an example of the same code with the including values of speed, weight, color, and disappearance.
+
+```
+// Creates one fleeting red spiral and one blue spiral
+Rule spiralRule = 
+{
+	Right: 10
+	Down: 10
+	Left: 15
+	Up: 15
+	Right: 20
+	Down: 20
+	Left: 25
+	Up: 25
+}
+
+Speed(20)
+Weight(50)
+Color(red)
+Disappearance(true)
+
+Start (50, 50)
+Do spiralRule
+
+Speed(20)
+Weight(50)
+Color(blue)
+	
+Start (100, 50)
+Do spiralRule
+```
 
 ## Operation (semantics)
 _What might happen when a program runs? Is there a computational model that
 corresponds with your domain? How does a program interact with the
 user? What kinds of semantic (i.e., non-syntax) errors might occur, and how 
 might they be communicated to the user?_
 
+This domain corresponds with a data interpretation model, since the user is basically 
+defining a set of rules results in the corresponding 2D line animation. More specifically, 
+it can also be seen as state interpretation because the user is defining the state of a 
+line or multiple lines and where they should travel next. When the program runs, the rules 
+will be read in line by line and the appropriate line/rectangle objects will be created. 
+A standalone GUI application will pop up that shows the 2D animation produced by their 
+rules. As the user is writing their code, they can intermittently run the program to have 
+a sense of what their rules are producing. 
+
+One semantic error that could occur is the line being out of boundary. In this case, the 
+user should be notified with an error message that says, “Out of boundary!”. Additionally, 
```

Comment:
Would you still draw the animation up to the boundary? i.e. if someone wrote `Right 10000` and started in a valid coordinate, would the animation run and throw an error at the boundary, or would you throw an error before the animation starts?